### PR TITLE
Add New Rule CP.3

### PR DESCRIPTION
**Pull request by AmeenAli**
Add New Rule CP.3
(Consider void futures for one-shot event communication)
once again this is just  making effective use of the new capabilities of the language.

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -8060,6 +8061,22 @@ Typically, such programs leads to a painful effort to remove data races.
 
 **Enforcement**: Some is possible, do at least something.
 
+<a name="Rconc-voidf"></a>
+### CP.3: Consider void futures for one-shot event communication
```
Comment (tkruse) [inline CppCoreGuidelines.md]: please move the html anchor inside the header as has been done for all the headers on the master branch.

**Comment by cubbimew** [comment]
Is this really a good idea? Unlike condition variables, promises are not reusable and incur heap allocations for the shared state.
Also, speaking of modern facilities, there is latch(1) from Concurrency TS

**Comment by jaredgrubb** [comment]
Although latch does solve some parts of that problem, it's a synchronously-blocking mechanism. For an async system, future<void> is the best tool (especially with the TS features).

**Comment by hsutter** [comment]
The title and body are inconsistent: The title recommends `future<void>`, but the body doesn't mention `future` at all and only talks about `condition_variable`. Please explain?

**Comment by BjarneStroustrup** [comment]
"one short communication" needs more work

**Pull request closed without merging**